Zend/zend_objects.c

```diff
@@ -253,9 +262,18 @@ ZEND_API void ZEND_FASTCALL zend_objects_clone_members(zend_object *new_object,
 		} ZEND_HASH_FOREACH_END();
 	}
 
-	if (old_object->ce->clone) {
+	if (has_clone_method) {
 		GC_ADDREF(new_object);
 		zend_call_known_instance_method_with_0_params(new_object->ce->clone, new_object, NULL);
+
+		for (uint32_t i = 0; i < new_object->ce->default_properties_count; i++) {
+			zend_property_info *prop_info = new_object->ce->properties_info_table[i];
+			if (prop_info && (prop_info->flags & ZEND_ACC_READONLY)) {
```

Comment:
Why is this check for readonly needed here while it's not when setting the flag above? Shouldn't there be some symmetry, like set+unset flag for all props xor only readonly props?

Reply:
Also: wouldn't adding the same check to the previous lines provide more performance since it would prevent any writes to memory unless readonly props are found?

Reply:
I'm not sure anymore why I used a different approach 🤔 . Regarding performance, Tyson had some great ideas, so I'll try to reduce the performance impact of my changes according to his suggestions as I'm not exactly sure that the best approach would be. :)